src/main/java/org/haffson/adventofcode/days/day06/Coordinate.java

```diff
@@ -0,0 +1,358 @@
+                if (coordinateDistance.equals(shortestDistance)) {
+                    nearestCoordinates.add(comparedCoordinate);
+                }
+            });
+
+            return nearestCoordinates;
+        }
+
+        /**
+         * Helper method for day 6.
+         * Needs a Coordinate and compares it to given coordinates.
+         * Uses a helper method to get a List<Integer> of the closest coordinates.
+         *
+         * @return List<Integer> with coordinate ids.
+         */
+        @Nonnull
+        private List<Integer> findNearestCoordinates(@Nonnull final Coordinate coordinateToCompare) {
+            Map<Integer, Integer> distancesByCoordinate = getDistanceByCoordinate(coordinateToCompare);
+
+            return findShortestDistances(distancesByCoordinate);
+        }
+
+        /**
+         * Helper method for day 6.
+         * Needs a Coordinate and compares it to given coordinates.
+         * Maps coordinate distance by compared coordinate
+         *
+         * @return Map<Integer, Integer> distanceByCoordinate.
+         */
+        @Nonnull
+        private Map<Integer, Integer> getDistanceByCoordinate(@Nonnull final Coordinate coordinateToCompare) {
+            Map<Integer, Integer> distancesByCoordinate = new HashMap<>();
+            coordinates.forEach(coordinate -> distancesByCoordinate
+                    .put(coordinates.indexOf(coordinate), coordinateFieldDistance(coordinateToCompare, coordinate)));
+
+            return distancesByCoordinate;
+        }
+
+        /**
+         * Helper method for day 6.
+         * Needs a List<Integer> with Coordinate ids.
+         * Proves whether the list contains one or more values.
+         * If the list just contains one value, it means the spot in the coordinate system is only claimed by one coordinate.
+         * If the list contains two or more values, the spot is claimed by more coordinates.
+         * Is it claimed by more coordinates it should be marked with a ".".
+         *
+         * @return String to fill the coordinate system.
+         */
+        @Nonnull
+        private String givesCoordinateSystemInput(@Nonnull final List<Integer> nearestCoordinates) {
+            if (nearestCoordinates.size() > 1) {
+                return ".";
+            } else {
+                return String.valueOf(nearestCoordinates.get(0));
+            }
+        }
+
+        /**
+         * Helper method for day 6.
+         * Iterates over the coordinate system.
+         * Fills each entry with a coordinate id or "." when a spot is claimed by several coordinates
+         */
+        private void fillCoordinateSystem() {
+            Arrays.stream(coordinateSystem).forEach(arrayOfXCoordinates ->
+                    IntStream.range(0, arrayOfXCoordinates.length)
+                            .forEach(yCoordinate -> {
+                                int xCoordinate = Arrays.asList(coordinateSystem).indexOf(arrayOfXCoordinates);
```

Comment:
Brauchst du hier eigentlich nur den Index der X-Zeile?

Dann könntest du auch 2 ganz langweilige `for`-Schleifen benutzen. Dann kann man einfacher auf die Indexe zugreifen.

Reply:
Genau. Die Y-Koordinate bekomme ich direkt über `forEach` ausgegeben, aber die x-Koordinate ist etwas umständlicher zu bekommen. Der `for-loop` könnte hier tatsächlich Vorteile bieten und die Abfrage erleichtern.